docs/guides/validation-user-guide.md

```diff
@@ -4,4 +4,437 @@ headline: Validation User Guide
+only one field in the group can be set at a time. Instead, Spine provides `(is_required)` option:
+
+```proto
+import "spine/options.proto";
+
+message UserIdentity {
+    oneof auth_type {
+        option (is_required) = true;
+        
+        EmailAddress email = 1;
+        GoogleId google = 2; 
+        TwitterId twitter = 3; 
+    }
+}
+```
+
+In this case one of the fields `UserIdentity.email`, `UserIdentity.google`,
+and `UserIdentity.twitter` must be set.
+
+`(is_required)` option is not yet supported in Dart.
+
+Other cases call for conditional required fields. In particular, some fields of a message must be
+set alongside other fields. Consider an example of an online store item:
+
+```proto
+import "spine/options.proto";
+
+message Item {
+
+    // ...
+
+    Timestamp when_opened_for_sale = 42;
+    UserId who_opened_for_sale = 43 [(goes).with = "when_opened_for_sale"];
+}
+```
+
+The `Item.who_opened_for_sale` field only makes sense for the domain if 
+the `Item.when_opened_for_sale` field is set. If `who_opened_for_sale` is set and
+`when_opened_for_sale` is not, a constraint violation is produced.
+
+Finally, there are some cases, in which a pair of fields may be set at the same time, but at least
+one of them must be set. This and more complex cases are handled by the type-level
+`(required_field)` option:
+
+```proto
+import "spine/options.proto";
+
+message PersonName {
+    option (required_field) = "given_name|honorific_prefix & family_name";
+
+    string honorific_prefix = 1;
+    string given_name = 2;
+    string middle_name = 3;
+    string family_name = 4;
+    string honorific_suffix = 5;
+}
+``` 
+
+In case of `PersonName`, either `given_name` or both `honorific_prefix` and `family_name` must be
+set. All three can be set at the same time.
+
+#### Missing fields
+
+In case if a required field is missing, the validation error message will explicitly say so.
+However, if you need a specific error message for this field, you can provide it via
+the `(if_missing)` option:
+
+```proto
+import "spine/options.proto";
+
+message PhoneNumber {
+    string digits = 1 [(required) = true,
+                       (msg_format = "Phone number must contain digits.")];
+}
+```
+
+Note that this option only applies to fields marked with `(required)` and not to the fields
+referenced via any other options.
+
+If `(goes)` option is used, the error message can be customized with the `(goes).msg_format`
+parameter. Note that the message should contain two "`%s`" insertion points: first for the name of
+the field declaring the option and second for the name of the field targeted by the option.
+
+#### When `(required)` is implicit
+
+In Spine, by convention, if the first declared field of a [Command](../introduction/naming-conventions.html#command-definitions),
+an [Event](../introduction/naming-conventions.html#event-definitions), or an entity state is
```

Comment:
"an"? 

Let's rephrase this anyway. `When defining the domain commands, events or entity states, we have found to be convenient that the first field of the respective Message is the identifier. It's just easier to read. Therefore, by convention, Spine treats the first fields of such objects as their IDs: (example).` Then you say — `however if for some reason you have declared the repeated field first, ...`.

Notice, the positive straightforward, then its example, then exceptions and complications.